### PR TITLE
Support reading broker feature flags

### PR DESCRIPTION
**Pull request by oldalton**
## Proposed changes

Adding additional field for the SSO extension protocol that allows passing additional information/settings.
This is needed to allow Authenticator read settings from SSO extension.

## Type of change

- [x] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/util/NSJSONSerialization+MSIDExtensions.m [hunk 1]
```diff
@@ -28,6 +28,11 @@ @implementation NSJSONSerialization (MSIDExtensions)
 
 + (NSDictionary *)msidNormalizedDictionaryFromJsonData:(NSData *)data error:(NSError **)error
 {
+    if (!data)
```
Comment (hieunguyenmsft) [inline IdentityCore/src/util/NSJSONSerialization+MSIDExtensions.m:31]: nit: is it good practice to check both !data and [data length] == 0, just to be safe? I was updated in my PR  yesterday following similar Rohit's comment about that.

File: IdentityCore/src/broker_operation/response/MSIDDeviceInfo.m [hunk 2]
```diff
@@ -59,6 +60,9 @@ - (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(__unused NSEr
         _ssoExtensionMode = [self ssoExtensionModeEnumFromString:[json msidStringObjectForKey:MSID_BROKER_SSO_EXTENSION_MODE_KEY]];
         _wpjStatus = [self wpjStatusEnumFromString:[json msidStringObjectForKey:MSID_BROKER_WPJ_STATUS_KEY]];
         _brokerVersion = [json msidStringObjectForKey:MSID_BROKER_BROKER_VERSION_KEY];
+        
+        NSData *jsonData = [json[MSID_ADDITIONAL_EXTENSION_DATA_KEY] dataUsingEncoding:NSUTF8StringEncoding];
+        _additionalExtensionData = [NSJSONSerialization msidNormalizedDictionaryFromJsonData:jsonData error:nil];
```
Comment (jasoncoolmax) [inline IdentityCore/src/broker_operation/response/MSIDDeviceInfo.m:65]: This will crash on nil `jsonData`? :)

Comment (jasoncoolmax) [reply IdentityCore/src/broker_operation/response/MSIDDeviceInfo.m:65]: Oh, ignore my comment, you made a new check in that method :)

File: IdentityCore/src/util/NSJSONSerialization+MSIDExtensions.m (hunk 1 above)
Comment (kaisong1990) [reply IdentityCore/src/util/NSJSONSerialization+MSIDExtensions.m:31]: For my understanding, will ```[data length] == 0``` not always give 0 if data is nil?

Comment (hieunguyenmsft) [reply IdentityCore/src/util/NSJSONSerialization+MSIDExtensions.m:31]: Cool, thanks Kai for the tip!

Comment (oldalton) [reply IdentityCore/src/util/NSJSONSerialization+MSIDExtensions.m]: Addressed.

**Pull request merged**